pkg/ingester/ingester_v2.go

```diff
@@ -847,6 +855,36 @@ func (i *Ingester) v2Push(ctx context.Context, req *cortexpb.WriteRequest) (*cor
 				return copiedLabels
 			})
 		}
+
+		// app.AppendExemplar currently doesn't create the series, it must
+		// already exist.  If it does not then drop.
+		if ref == 0 && len(ts.Exemplars) > 0 {
```

Comment:
I would do this entire logic only if exemplars ingestion is enabled.

I'm looking in particular at the else branch. I know that `app.AppendExemplar()` is a no-op if `max-exemplars` is set to zero, but we would pay the cost of copying labels anyway even if exemplars are disabled in Cortex.